### ISSUE TITLE
请问这三个文件的区别？以及如何在程序中选择

### ISSUE THREAD

**Issue by Wanghe1997** [ultralytics/yolov5#5653]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

![1636981175(1)](https://user-images.githubusercontent.com/45507837/141786214-70b3142f-ee5a-4597-b857-482390c5e574.png)
作者您好，请问hyp.scratch这4个yaml文件有什么区别？各自是在什么样的情况下使用？程序应该在哪里选择使用哪个hyp.scratch文件呢？另外，标注红色方框的这两个yaml文件又有什么区别？

### Additional

像这些细节性的问题（比如this question，或者是一些超参数的含义等等诸如此类细节性的问题）我好像在你们的tutorial里找不到相关的解释，是不是有些关于程序或者参数设置的问题你们在一些issue里面给出了答案呢？有时候我有一些关于你们程序参数的设置、含义等疑问不知道从哪里能找到解答。你们是否在一些issue里面整理了一些常见的疑问，如果有的话可以分享吗？

**Comment by Wanghe1997** [comment]
PS：Because I am from China, it is more convenient and time-saving to ask questions by using Chinese. I don’t know if you can understand the meaning of my question. If it’s not easy for you to understand, I will continue to ask questions in English in the future. lol

**Comment by Wanghe1997** [comment]
Your tutorial, I feel that the writing is more general and broad, some detailed questions about the meaning of the parameters/files and how to configure them, where should I look for answers? Are there related issues that you sorted out or summarized？Thanks

**Comment by glenn-jocher** [comment]
@Wanghe1997 hyp files are specified during training here, i.e. `python train.py --hyp hyp.scratch.yaml`
https://github.com/ultralytics/yolov5/blob/c2523be634a94da2b1b2a43c11b25827a0de990d/train.py#L445

- finetune.yaml: evolved on VOC
- finetune_objects365: evolved on Objects365
- scratch: default
- scratch_low: for smaller models, i.e. v5n, v5s
- scratch_med: for medium models, i.e. v5m
- scratch_high: for large models, i.e. v3, v3-spp, v5l, v5x

**Comment by Wanghe1997** [comment]
> @Wanghe1997 hyp files are specified during training here, i.e. `python train.py --hyp hyp.scratch.yaml`
> 
> https://github.com/ultralytics/yolov5/blob/c2523be634a94da2b1b2a43c11b25827a0de990d/train.py#L445
> 
> * finetune.yaml: evolved on VOC
> * finetune_objects365: evolved on Objects365
> * scratch: default
> * scratch_low: for smaller models, i.e. v5n, v5s
> * scratch_med: for medium models, i.e. v5m
> * scratch_high: for large models, i.e. v3, v3-spp, v5l, v5x

Thank u. 
Q1：Does scratch: default mean that all size models can be tried？Why should a small model set a small initial learning rate?
Q2：For small models（v5n,v5s）, is it better to set the batch-size to a small value（8,16） or a large value(32,64,128) during the training process? Are there any techniques or arguments for the batch-size setting?

**Comment by glenn-jocher** [comment]
@Wanghe1997 batch size should be set as large as possible to fully utilize your GPU. 

### AutoBatch

You can use [YOLOv5 AutoBatch](https://github.com/ultralytics/yolov5/pull/5092) (NEW) to find the best batch size for your training by passing `--batch-size -1`. AutoBatch will solve for a 90% CUDA memory-utilization batch-size given your training settings. AutoBatch is experimental, and only works for Single-GPU training. It may not work on all systems, and is not recommended for production use.

<img width="1746" alt="Screenshot 2021-11-06 at 12 31 10" src="https://user-images.githubusercontent.com/26833433/140608327-9c0c0baf-a0ee-4551-a4ff-5ee26b155c51.png">

Good luck and let us know if you have any other questions!

**Comment by Wanghe1997** [comment]
> @Wanghe1997 batch size should be set as large as possible to fully utilize your GPU.
> 
> ### AutoBatch
> You can use [YOLOv5 AutoBatch](https://github.com/ultralytics/yolov5/pull/5092) (NEW) to find the best batch size for your training by passing `--batch-size -1`. AutoBatch will solve for a 90% CUDA memory-utilization batch-size given your training settings. AutoBatch is experimental, and only works for Single-GPU training. It may not work on all systems, and is not recommended for production use.
> 
> <img alt="Screenshot 2021-11-06 at 12 31 10" width="1746" src="https://user-images.githubusercontent.com/26833433/140608327-9c0c0baf-a0ee-4551-a4ff-5ee26b155c51.png">
> 
> Good luck and let us know if you have any other questions!
So according to the results of this picture, the best batch-size of the A100-SXM4 graphics card for training with 640 img-size on the coco128 data set is 179? Is my understanding correct?

**Comment by glenn-jocher** [comment]
@Wanghe1997 yes

**Comment by Wanghe1997** [comment]
> AutoBatch


So just set batch-size to 1 in train.py, will the program automatically execute AutoBatch? No need to do any configuration other than that?

**Comment by glenn-jocher** [comment]
@Wanghe1997 no, the screenshot shows -1, not 1

**Comment by Wanghe1997** [comment]
> @Wanghe1997 no, the screenshot shows -1, not 1

ok，thanks

**Comment by Wanghe1997** [comment]
> @Wanghe1997 no, the screenshot shows -1, not 1

I just tested AutoBatch with my dataset, and the best batch-size measured was 111. However, I used this value of 111 to train yolov5s, and it prompted an error: Unable to find a valid cuDNN algorithm to run convolution. I feel that the Autobatch test is not accurate! Although my graphics card is RTX3090, 24G memory, but theoretically it can't bear such a large batch-size

**Comment by glenn-jocher** [comment]
@Wanghe1997 thank you for your feedback. As noted above:
<img width="879" alt="Screenshot 2021-11-19 at 14 50 40" src="https://user-images.githubusercontent.com/26833433/142633320-9a2d7ef1-378f-43db-b7db-8341df73e252.png">

**Comment by Wanghe1997** [comment]
> @Wanghe1997 thank you for your feedback. As noted above: <img alt="Screenshot 2021-11-19 at 14 50 40" width="879" src="https://user-images.githubusercontent.com/26833433/142633320-9a2d7ef1-378f-43db-b7db-8341df73e252.png">

You're welcome. My system is windows10, it happens to be tested with RTX3090 single card. It seems that AutoBatch is not accurate yet. Or it can only be calculated for coco128, not for custom datasets?

**Comment by glenn-jocher** [comment]
@Wanghe1997 the dataset does not matter. It's only been evaluated on Colab instances with K80, T4, P100, V100, A100, so no consumer cards yet.
